rclcpp/test/test_function_traits.cpp

```diff
@@ -0,0 +1,348 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <rclcpp/function_traits.hpp>
+
+int func_no_args()
+{
+  return 0;
+}
+
+int func_one_int(int x)
+{
+  (void)x;
```

Comment:
If we never intend to document the function we could remove the variable name from the signature and then also this line.